File: Top 100 Codes/02-Working with Numbers/04-Binary to Decimal to conversion/BinarytoDecimaltoconversion.cpp
//using built-in stoi() function
#include <iostream>
#include <string>
#include <exception>
using namespace std;

int main()
{
    string binaryNumber;
    cout << "Enter a binary number: ";
    cin >> binaryNumber;

    try {
        // Convert the binary string to a decimal number
        int decimalValue = stoi(binaryNumber, 0, 2);
        cout << "Decimal value: " << decimalValue << endl;
    } catch (const invalid_argument &e) {
        cout << "Error: Invalid binary number. Please enter only 0s and 1s." << endl;
    } catch (const out_of_range &e) {
        cout << "Error: The binary number is too large to convert." << endl;
    }

    return 0;
}

File: Top 100 Codes/02-Working with Numbers/04-Binary to Decimal to conversion/BinarytoDecimaltoconversion.py
# Input: binary number as a string
binary_number = input("Enter a binary number: ")

try:
    # Convert binary to decimal using int() with base 2
    decimal_number = int(binary_number, 2)
    print(f"Decimal value: {decimal_number}")
except ValueError:
    print("Error: Invalid binary number. Please enter only 0s and 1s.")
